shared/src/hover/HoverOverlay.tsx

```diff
@@ -33,6 +34,28 @@ export interface HoverOverlayClassProps {
     actionItemPressedClassName?: string
 }
 
+/**
+ * A dismissable alert to be displayed in the hover overlay.
+ */
+export interface HoverAlert<T extends string = string> {
```

Comment:
```suggestion
export interface HoverAlert<T extends string> {
```

Let's enforce type safety